src/kernel/norm_fiber/cuda.cu

```diff
@@ -0,0 +1,110 @@
+/*! @copyright (c) 2022-present Skolkovo Institute of Science and Technology
+ *                              (Skoltech), Russia. All rights reserved.
+ *                 2023-present Artificial Intelligence Research Institute
+ *                              (AIRI), Russia. All rights reserved.
+ *
+ * NNTile is software framework for fast training of big neural networks on
+ * distributed-memory heterogeneous systems based on StarPU runtime system.
+ *
+ * @file src/kernel/norm_fiber/cuda.cu
+ * Sums over slices into a fiber of a buffer on CUDA
+ *
+ * @version 1.0.0
+ * */
+
+#include "nntile/kernel/norm_fiber/cuda.hh"
+#include <cmath>
+#include <algorithm>
+#include "nntile/kernel/cuda.hh"
+
+namespace nntile::kernel::norm_fiber
+{
+
+template<typename T>
+static __global__
+void cuda_kernel(Index m, Index n, Index k, Index batch, Scalar alpha, const T *src,
+        Scalar beta, T *dst)
+//! Sums over slices along the first and last axes into a fiber of a tensor
+/*! For a provided m-by-k-by-n input array computes sums over slices
+ * along the first axis with m elements and the last axis with n elements,
+ * resulting in output fiber of shape (k).
+ * Mnemonically, the following operations are performed:
+ *      dst[l,b] = hypot(beta*dst[l,b], alpha*norm(src[:,l,:,b]))
+ *
+ * @param[in] m: Size of the first mode of src array
+ * @param[in] n: Size of the last mode of src array
+ * @param[in] k: Size of the middle mode of src array and the only mode of
+ *      dst array
+ * @param[in] batch: Size of the batch dimension
+ * @param[in] alpha: Scaling factor for src
+ * @param[in] src: Input contiguous m-by-k-by-n array
+ * @param[in] beta: Scaling factor for dst
+ * @param[inout] dst: Output contiguous vector with k elements, that accumulate
+ *      norm over slices along the first and the last axes.
+ * */
+{
+    /* Index i2_batched = threadIdx.x + blockIdx.x*blockDim.x;
+    Index i2 = i2_batched % k;
+    Index b = i2_batched / k;
+    Index i0_start = threadIdx.y, i0_step = blockDim.y;
+    Index i1_start = threadIdx.z, i1_step = blockDim.z;
+    using Y = typename T::repr_t;
+    constexpr Y zero{0.0};
+    // Init sum
+    Y sum = zero;
+    */
+    // this is placholder - TODO
```

Comment:
cuda code does nothing, is it intended?

Reply:
Yes - as we disused cuda will be added later